kaal-infrastructure/src/main/kotlin/cz/eman/kaal/infrastructure/api/retrofit/KaalRetrofitCaller.kt

```diff
@@ -27,6 +28,13 @@ import java.util.Optional
 @Suppress("unused")
 open class KaalRetrofitCaller {
 
+    /**
+     * The list of exceptions to re-throw from [callResult], [callResultOptional] and [callResultCompleteOptional]
+     * methods. Could be overridden in implementations to provide a list of exception classes which should be re-thrown.
+     * By default list is empty and each exception is wrapped in [Result.Error].
+     */
+    open val rethrowExceptions: List<KClass<*>> = emptyList()
```

Comment:
Is there any potential exception other than CancellationException that could be thrown again?

Reply:
Theoretically nope, but who knows 🤷‍♂️ 

My first idea was to add only `rethrowCancellationException: Boolean = false` to the `KaalRetrofitCaller`, but it was too rigid for me. Alternatively we can just open `handleCallException` method see comment: https://github.com/eManPrague/kaal/issues/57#issuecomment-1892601835

Reply:
Okey, my preferences are:
- re-throw `CancellationException` by default
- make handleCallException as `open`

Reply:
Done 👌